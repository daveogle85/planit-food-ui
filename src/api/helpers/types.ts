export enum RestVerb {
  GET = 'GET',
  POST = 'POST',
  PUT = 'PUT',
  DELETE = 'DELETE',
  OPTIONS = 'OPTIONS',
}

export type Options = {
  method: RestVerb;
  redirect: 'follow' | 'error' | 'manual' | undefined;
  headers: Headers;
  body?: string;
};
