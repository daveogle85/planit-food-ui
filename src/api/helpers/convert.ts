import { v4 as uuid } from 'uuid';

import { ApiDish, Dish } from '../types/DishTypes';
import { ApiMeal, Meal } from '../types/MealTypes';
import { List, ApiList } from '../types/ListTypes';
import { ApiDay, Day } from '../types/DayTypes';

export const convertFromDishApi = (dish: ApiDish): Dish => ({
  ...dish,
  localId: uuid(),
});

export const convertFromMealApi = (meal: ApiMeal): Meal => {
  const dishes = meal.dishes?.map(d => {
    const dish = convertFromDishApi(d);
    return dish;
  });
  return { ...meal, localId: uuid(), dishes };
};

export const convertFromDayApi = (day: ApiDay): Day => {
  const meal = convertFromMealApi(day.meal);
  return { ...day, meal };
};

export const convertToApiDish = (dish: Dish): ApiDish => {
  const { localId, ...apiDish } = dish;
  return apiDish;
};

export const convertToLightApiDish = (dish: Dish): ApiDish => {
  const { id, name, notes, dishType } = dish;
  return {
    id,
    name,
    notes,
    dishType,
  };
};

export const convertToApiMeal = (meal: Meal): ApiMeal => {
  const { localId, ...apiMeal } = meal;
  return {
    ...apiMeal,
    dishes: (meal.dishes ? meal.dishes.map(convertToApiDish) : []) as ApiDish[],
  };
};

export const convertToLightApiMeal = (meal: Meal): ApiMeal => {
  const { id, name, dishes, notes } = meal;
  const updatedDishes = dishes?.map(convertToLightApiDish);
  return {
    id,
    name,
    dishes: updatedDishes,
    notes,
  };
};

export const convertToListApiRequest = (list: List): ApiList => {
  const updatedMeals = list.meals.map(convertToLightApiMeal);
  return {
    ...list,
    meals: updatedMeals,
  };
};

export const convertToLightApiDay = (day: Day): ApiDay => {
  const meal = convertToLightApiMeal(day.meal);
  return { ...day, date: day.date!, meal };
};

export const convertToApiDay = (day: Day): ApiDay => {
  const meal = convertToApiMeal(day.meal);
  return { ...day, date: day.date!, meal };
};
