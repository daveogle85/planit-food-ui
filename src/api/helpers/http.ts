async function fetchWithTimeout(
  url: string,
  options = {},
  timeout = 5000
): Promise<Response> {
  const controller = new AbortController();
  const { signal } = controller;
  return new Promise((resolve, reject) => {
    const timer = setTimeout(() => {
      reject(new Error(`Timeout for request to ${url}`));
      controller.abort();
    }, timeout);
    fetch(url, { signal, ...options })
      .finally(() => clearTimeout(timer))
      .then(resolve, reject);
  });
}

export async function get<T>(path: string, accessToken: string): Promise<T> {
  try {
    const response = await fetchWithTimeout(
      `${process.env.REACT_APP_SERVER_BASE_URL}${path}`,
      {
        method: 'GET',
        headers: new Headers({
          'Content-Type': 'application/json',
          authorization: `Bearer ${accessToken}`,
        }),
      }
    );
    return response.json();
  } catch (ex) {
    console.error(ex.message);
    throw ex;
  }
}

// async function post<T>(url: string, body: T) {
//   try {
//     const { getIdTokenClaims } = useAuth0();
//     const accessToken = await getIdTokenClaims();
//     const response = await fetch(
//       `${process.env.REACT_APP_SERVER_BASE_URL}/blog/post`,
//       {
//         method: 'post',
//         headers: new Headers({
//           'Content-Type': 'application/json',
//           Accept: 'application/json',
//           authorization: `Bearer ${accessToken.__raw}`,
//         }),
//         body: JSON.stringify(body),
//       }
//     );
//     return response.ok;
//   } catch (ex) {
//     return false;
//   }
// }
