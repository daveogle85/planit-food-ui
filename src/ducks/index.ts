import { combineReducers, Action, ThunkAction } from '@reduxjs/toolkit';
import authReducer from './auth/AuthReducer';
import daysReducer from './days/DaysReducer';
import listsReducer from './lists/ListsReducer';
import toastReducer from './toast/ToastReducer';
import calendarReducer from './calendar/CalendarReducer';

const rootReducer = combineReducers({
  auth: authReducer,
  calendar: calendarReducer,
  days: daysReducer,
  lists: listsReducer,
  toast: toastReducer,
});

export type RootState = ReturnType<typeof rootReducer>;
export type AppThunk = ThunkAction<void, RootState, unknown, Action<string>>;

export default rootReducer;
