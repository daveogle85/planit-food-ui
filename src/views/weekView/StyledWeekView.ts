import styled from '../../styles/theme';

export const styledWeekView = (WeekView: React.ComponentType) => styled(
  WeekView
)`
  height: 100%;
  background-color: ${props => props.theme.colours.background.weekViewPurple};
  background-image: linear-gradient(
    315deg,
    ${props => props.theme.colours.background.weekViewPurple} 0%,
    ${props => props.theme.colours.background.weekViewBlue} 74%
  );
`;
