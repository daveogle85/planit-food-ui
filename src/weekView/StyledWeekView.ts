import styled from '../styles/theme';
import { fullScreenRelative } from '../styles/common';
import { navBar } from '../styles/heights';

export const styledWeekView = (WeekView: React.ComponentType) => styled(
  WeekView
)`
  ${fullScreenRelative(navBar)}
`;
