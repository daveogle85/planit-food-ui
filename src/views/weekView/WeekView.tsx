import classNames from 'classnames';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { RootState } from '../../ducks';
import {
  fetchDayDataForCarousel,
  daysSelectors,
} from '../../ducks/days/DaysReducer';
import { EmotionProps } from '../../styles/types';
import DayCardCarousel from '../../components/Carousel/DayCardCarousel';
import NavBar from '../../components/NavBar/NavBar';
import { styledWeekView } from './StyledWeekView';
import Loading from '../../components/Loading/Loading';

const WeekView: React.FC<EmotionProps> = props => {
  const dispatch = useDispatch();
  const loadingDays = useSelector(daysSelectors.selectLoading);
  const days = useSelector(daysSelectors.selectData);
  const token = useSelector((state: RootState) => state.auth.token);

  useEffect(() => {
    async function fetchDays() {
      dispatch(fetchDayDataForCarousel());
    }

    fetchDays();
  }, [dispatch, token]);
  return (
    <>
      <NavBar />
      <div className={classNames('WeekView', props.className)}>
        <Loading isLoading={loadingDays}>
          <DayCardCarousel days={days} />
        </Loading>
      </div>
    </>
  );
};

export default styledWeekView(WeekView);
