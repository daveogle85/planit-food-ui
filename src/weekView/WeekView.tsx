import classNames from 'classnames';
import React, { useEffect } from 'react';
import { useDispatch, useSelector } from 'react-redux';

import { RootState } from '../ducks';
import { fetchDayDataForCarousel, selectors } from '../ducks/days/DaysReducer';
import { EmotionProps } from '../styles/types';
import DayCardCarousel from '../components/Carousel/DayCardCarousel';
import NavBar from '../components/NavBar/NavBar';
import { styledWeekView } from './StyledWeekView';
import LoadingSpinner from '../components/Spinner/Spinner';

const WeekView: React.FC<EmotionProps> = props => {
  const dispatch = useDispatch();
  const loadingDays = useSelector(selectors.selectLoading);
  const days = useSelector(selectors.selectWeek);
  const token = useSelector((state: RootState) => state.auth.token);

  useEffect(() => {
    async function fetchDays() {
      dispatch(fetchDayDataForCarousel());
    }

    fetchDays();
  }, [dispatch, token]);
  return (
    <>
      <NavBar />
      <div className={classNames('WeekView', props.className)}>
        {loadingDays ? <LoadingSpinner /> : <DayCardCarousel days={days} />}
      </div>
    </>
  );
};

export default styledWeekView(WeekView);
