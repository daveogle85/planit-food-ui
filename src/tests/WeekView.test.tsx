import serializer from 'jest-emotion';
import React from 'react';

import { styledWeekView } from '../weekView/StyledWeekView';
import WeekView from '../weekView/WeekView';
import { renderWithTheme } from './helpers';
import { RootState } from '../ducks';

jest.mock('../contexts/auth0-context', () => ({
  useAuth0: jest.fn().mockReturnValue({
    loginWithRedirect: jest.fn(),
    loading: false,
    isAuthenticated: false,
    user: {
      picture: '',
    },
  }),
}));

expect.addSnapshotSerializer(serializer);
test('Week View renders correctly', () => {
  const mockStore: RootState = {
    days: {
      loading: false,
      data: [],
      week: [],
    },
    auth: { isAuthenticated: true, token: '' },
  };
  const TestWeekView = styledWeekView(WeekView as any);
  const view = renderWithTheme(<TestWeekView />, mockStore).toJSON();
  expect(view).toMatchSnapshot();
});
