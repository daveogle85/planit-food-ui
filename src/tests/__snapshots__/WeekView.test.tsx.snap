// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Week View renders correctly 1`] = `
.emotion-12 {
  height: 100%;
  background-color: #7f53ac;
  background-image: linear-gradient( 315deg, #7f53ac 0%, #647dee 74% );
  height: 100%;
  background-color: #7f53ac;
  background-image: linear-gradient( 315deg, #7f53ac 0%, #647dee 74% );
}

.emotion-13 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  top: 70px;
}

.emotion-10 {
  position: fixed;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: #fff;
  top: 0;
  left: 0;
  right: 0;
  padding: 0;
  height: 70px;
  box-shadow: 4px 0 12px -6px #000, -4px 0 12px -6px #000;
  list-style-type: none;
  z-index: 3;
}

.emotion-10 ul {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  position: relative;
  width: 100%;
  margin: 0;
  padding-left: 16px;
}

.emotion-10 ul li {
  cursor: pointer;
  list-style-type: none;
  height: 100%;
  min-width: 30px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-10 ul li:hover:not(.app-title) {
  background-color: #f1f1f1;
}

.emotion-10 ul .title-text {
  display: none;
}

@media (min-width:418px) {
  .emotion-10 ul {
    padding-left: 40px;
  }

  .emotion-10 ul .title-text {
    display: block;
  }
}

.emotion-10 ul .logo-img {
  width: 70px;
  height: 70px;
}

.emotion-10 ul .app-title {
  white-space: nowrap;
}

.emotion-10 ul .dd-menu-items {
  position: relative;
  height: calc(100% - 0px);
  width: calc(100% - 0px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: end;
  -webkit-justify-content: flex-end;
  -ms-flex-pack: end;
  justify-content: flex-end;
}

.emotion-10 ul .dd-menu-items li:not(.dd-list-item) {
  margin-right: 8px;
}

.emotion-2 {
  height: 100%;
  position: relative;
}

.emotion-2 .dd-header {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 100%;
}

.emotion-2 .dd-header-title {
  cursor: pointer;
  padding: 0 8px;
}

.emotion-2 li:hover {
  background-color: #46627f;
}

.emotion-2 .dd-list {
  z-index: 1;
  background-color: #fff;
  position: absolute;
  padding: 0;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  display: none;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  width: -webkit-max-content;
  width: -moz-max-content;
  width: max-content;
  min-width: 100%;
  border: 1px solid #2b2d2f;
  border-radius: 0.3em;
}

.emotion-2 .dd-list.dd-open {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-2 .dd-list.dd-oob {
  right: 0;
}

.emotion-2 .dd-list li {
  cursor: pointer;
  border-radius: 0.3em;
  padding: 0 8px;
  width: -webkit-fill-available;
  width: -moz-available;
  width: fill-available;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  height: 30px;
  line-height: 30px;
}

.emotion-1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  margin: 0 8px;
}

.emotion-1 .button-text {
  display: none;
}

.emotion-1 svg {
  position: relative;
  display: inline-block;
}

@media (min-width:768px) {
  .emotion-1 .button-text {
    display: block;
  }
}

.emotion-0 {
  border: 2px solid #9b9797;
  width: 40px;
  height: 40px;
  border-radius: 2em;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

@media (min-width:768px) {
  .emotion-0 {
    margin-right: 8px;
    width: 50px;
    height: 50px;
  }
}

.emotion-0 svg {
  width: 25px;
  height: 25px;
}

.emotion-8 {
  cursor: pointer;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding: 0 8px;
}

.emotion-8 div {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-8 .details {
  display: none;
  font-weight: 300;
  font-size: 15px;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

@media (min-width:768px) {
  .emotion-8 .details {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

.emotion-8 img {
  width: 40px;
  height: 40px;
  border-radius: 2em;
}

@media (min-width:768px) {
  .emotion-8 img {
    margin-right: 8px;
    width: 50px;
    height: 50px;
  }
}

.emotion-11 {
  position: absolute;
  top: 0;
  bottom: 0;
  left: 0;
  right: 0;
  margin: 20px;
  border-radius: 0.3em;
}

.emotion-11 .react-multi-carousel-track {
  height: 100%;
}

.emotion-11 .react-multi-carousel-item {
  height: 90%;
}

.emotion-11 .react-multiple-carousel__arrow {
  bottom: 0;
}

.emotion-11 .react-multiple-carousel__arrow--right {
  right: 0;
}

.emotion-11 .react-multiple-carousel__arrow--left {
  left: 0;
}

<div
  className="emotion-13 planit-food-pg"
>
  <nav
    className="emotion-10"
  >
    <ul
      className="menu"
    >
      <li
        className="app-title"
        onClick={[Function]}
      >
        <img
          alt="app logo"
          className="logo-img"
          src="logo.png"
        />
        <div
          className="title-text button-text"
        >
          Planit Food
        </div>
      </li>
      <div
        className="dd-menu-items"
      >
        <li
          className="dd-edit"
        >
          <div
            className="dd-wrapper emotion-2"
          >
            <div
              className="dd-header"
              onClick={[Function]}
            >
              <div
                className="emotion-1"
                title="View Edit Options"
              >
                <div
                  className="emotion-0"
                >
                  <svg
                    fill="#9b9797"
                    height="26px"
                    viewBox="0 0 26 26"
                    width="26px"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      d="M 20.09375 0.25 C 19.5 0.246094 18.917969 0.457031 18.46875 0.90625 L 17.46875 1.9375 L 24.0625 8.5625 L 25.0625 7.53125 C 25.964844 6.628906 25.972656 5.164063 25.0625 4.25 L 21.75 0.9375 C 21.292969 0.480469 20.6875 0.253906 20.09375 0.25 Z M 16.34375 2.84375 L 14.78125 4.34375 L 21.65625 11.21875 L 23.25 9.75 Z M 13.78125 5.4375 L 2.96875 16.15625 C 2.71875 16.285156 2.539063 16.511719 2.46875 16.78125 L 0.15625 24.625 C 0.0507813 24.96875 0.144531 25.347656 0.398438 25.601563 C 0.652344 25.855469 1.03125 25.949219 1.375 25.84375 L 9.21875 23.53125 C 9.582031 23.476563 9.882813 23.222656 10 22.875 L 20.65625 12.3125 L 19.1875 10.84375 L 8.25 21.8125 L 3.84375 23.09375 L 2.90625 22.15625 L 4.25 17.5625 L 15.09375 6.75 Z M 16.15625 7.84375 L 5.1875 18.84375 L 6.78125 19.1875 L 7 20.65625 L 18 9.6875 Z"
                    />
                  </svg>
                </div>
                <div
                  className="button-text"
                >
                  Edit
                </div>
              </div>
            </div>
            <div>
              <ul
                className="dd-list"
              >
                <li
                  className="dd-list-item"
                  onClick={[Function]}
                >
                  Edit A Meal
                </li>
                <li
                  className="dd-list-item"
                  onClick={[Function]}
                >
                  Edit A Dish
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          className="dd-calendar"
        >
          <div
            className="emotion-1"
            onClick={[Function]}
            title="Go To Calendar"
          >
            <div
              className="emotion-0"
            >
              <svg
                fill="#9b9797"
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M9 11H7v2h2v-2zm4 0h-2v2h2v-2zm4 0h-2v2h2v-2zm2-7h-1V2h-2v2H8V2H6v2H5c-1.11 0-1.99.9-1.99 2L3 20c0 1.1.89 2 2 2h14c1.1 0 2-.9 2-2V6c0-1.1-.9-2-2-2zm0 16H5V9h14v11z"
                />
              </svg>
            </div>
            <div
              className="button-text"
            >
              Calendar
            </div>
          </div>
        </li>
        <li
          className="dd-add"
        >
          <div
            className="dd-wrapper emotion-2"
          >
            <div
              className="dd-header"
              onClick={[Function]}
            >
              <div
                className="emotion-1"
                title="View Add Options"
              >
                <div
                  className="emotion-0"
                >
                  <svg
                    id="svg1"
                    version="1.0"
                    viewBox="0 0 60 60"
                    x="0"
                    y="0"
                  >
                    <path
                      d="m23.78 23.78v-13.999c0-1.9588 0.536-3.55 1.607-4.7741 1.071-1.2546 2.601-1.8819 4.59-1.8819s3.519 0.6273 4.59 1.8819c1.102 1.2241 1.653 2.8153 1.653 4.7737v13.999h13.816c2.019 0 3.611 0.551 4.773 1.653 1.194 1.071 1.791 2.586 1.791 4.544 0 1.989-0.597 3.534-1.791 4.636-1.162 1.102-2.754 1.652-4.773 1.652h-13.816v13.954c0 1.989-0.551 3.596-1.653 4.82-1.101 1.224-2.632 1.836-4.59 1.836s-3.488-0.612-4.59-1.836c-1.071-1.224-1.607-2.831-1.607-4.82v-13.954h-13.816c-1.9582 0-3.5494-0.596-4.7735-1.79-1.1934-1.224-1.7901-2.723-1.7901-4.498 0-1.958 0.5814-3.473 1.7442-4.544 1.1935-1.102 2.8-1.653 4.8196-1.653h13.816z"
                      fill="none"
                      id="path1092"
                      stroke="#ffffff"
                      strokeLinecap="round"
                      strokeLinejoin="round"
                      strokeWidth={5}
                    />
                    <path
                      d="m23.78 23.78v-13.999c0-1.9588 0.536-3.55 1.607-4.7741 1.071-1.2546 2.601-1.8819 4.59-1.8819s3.519 0.6273 4.59 1.8819c1.102 1.2241 1.653 2.8153 1.653 4.7737v13.999h13.816c2.019 0 3.611 0.551 4.773 1.653 1.194 1.071 1.791 2.586 1.791 4.544 0 1.989-0.597 3.534-1.791 4.636-1.162 1.102-2.754 1.652-4.773 1.652h-13.816v13.954c0 1.989-0.551 3.596-1.653 4.82-1.101 1.224-2.632 1.836-4.59 1.836s-3.488-0.612-4.59-1.836c-1.071-1.224-1.607-2.831-1.607-4.82v-13.954h-13.816c-1.9582 0-3.5494-0.596-4.7735-1.79-1.1934-1.224-1.7901-2.723-1.7901-4.498 0-1.958 0.5814-3.473 1.7442-4.544 1.1935-1.102 2.8-1.653 4.8196-1.653h13.816z"
                      fill="#9b9797"
                      id="path837"
                    />
                  </svg>
                </div>
                <div
                  className="button-text"
                >
                  Add
                </div>
              </div>
            </div>
            <div>
              <ul
                className="dd-list"
              >
                <li
                  className="dd-list-item"
                  onClick={[Function]}
                >
                  Add A Meal
                </li>
              </ul>
            </div>
          </div>
        </li>
        <li
          className="dd-profile"
        >
          <div
            className="dd-wrapper emotion-2"
          >
            <div
              className="dd-header"
              onClick={[Function]}
            >
              <div
                className="emotion-8"
                title="Logout"
              >
                <img
                  alt="Avatar"
                  src=""
                />
                <div
                  className="details"
                >
                  <span />
                  <span />
                </div>
              </div>
            </div>
            <div>
              <ul
                className="dd-list"
              >
                <li
                  className="dd-list-item"
                  onClick={[Function]}
                >
                  Logout
                </li>
              </ul>
            </div>
          </div>
        </li>
      </div>
    </ul>
  </nav>
  <div
    className="WeekView emotion-12"
  >
    <div
      className="react-multi-carousel-list carousel-container emotion-11"
    >
      <ul
        className="react-multi-carousel-track "
        onMouseDown={[Function]}
        onMouseEnter={[Function]}
        onMouseLeave={[Function]}
        onMouseMove={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchMove={[Function]}
        onTouchStart={[Function]}
        style={
          Object {
            "overflow": "unset",
            "transform": "translate3d(0px,0,0)",
            "transition": "none",
          }
        }
      />
    </div>
  </div>
</div>
`;
